features/networking/multus.feature

```diff
@@ -531,3 +531,78 @@ Feature: Multus-CNI related scenarios
+    # Make sure that the multus is enabled
+    Given the multus is enabled on the cluster
+    # Create the net-attach-def via cluster admin and simulating syntax errors
+    Given I have a project
+    When I run oc create as admin over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/NetworkAttachmentDefinitions/macvlan-bridge.yaml" replacing paths:
+      | ["metadata"]["name"] | macvlan-bridge-21756 |
+      | ["spec"]["config"]   | 'asdf'               |
+    Then the step should fail
+    And the output should contain:
+      | admission webhook "multus-validating-config.k8s.io" denied the request|
+    And admin ensures "macvlan-bridge-21756" network_attachment_definition is deleted from the "default" project after scenario
+    When I run oc create as admin over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/NetworkAttachmentDefinitions/macvlan-bridge.yaml" replacing paths:
+      | ["metadata"]["name"] | macvlan-bridge@$ |
+    Then the step should fail
+    And the output should contain:
+      |subdomain must consist of lower case alphanumeric characters|
+    And admin ensures "macvlan-bridge@$" network_attachment_definition is deleted from the "default" project after scenario
+    
+  # @author anusaxen@redhat.com
+  # @case_id OCP-21949
+  @admin
+  Scenario: The multus admission controller should be able to detect the issue in the pod template
+    # Make sure that the multus is enabled
+    Given the multus is enabled on the cluster
+    # Create the net-attach-def via cluster admin
+    Given I have a project
+    When I run oc create as admin over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/NetworkAttachmentDefinitions/macvlan-bridge.yaml"" replacing paths:
+      | ["metadata"]["name"] | macvlan-bridge-21456 |
+    Then the step should succeed 
+    And admin ensures "macvlan-bridge-21456" network_attachment_definition is deleted from the "default" project after scenario
+    # Create a pod consuming net-attach-def simulating wrong syntax in name
+    When I run oc create as admin over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/Pods/1interface-macvlan-bridge.yaml" replacing paths:
+      | ["metadata"]["generateName"] | macvlan-bridge-pod-$@ |
+    Then the step should fail
+    And the output should contain:
+      | subdomain must consist of lower case alphanumeric characters |
+    
+  # @author anusaxen@redhat.com
+  # @case_id OCP-21793
+  @admin
+  Scenario: User cannot consume the net-attach-def created in other project which is namespace isolated	
+    # Make sure that the multus is enabled
+    Given the multus is enabled on the cluster
+    # Create the net-attach-def via cluster admin
+    Given I switch to cluster admin pseudo user
+    And I use the "default" project
```

Comment:
@anuragthehatter do you have to use the `default` system project?

Reply:
@pruan-rht Yes this is multus feature. only a net-attach-def created under default project can be used by other projects if flag is enabled. So here we want to makes sure, with flag being disabled, the net-attach-def from default project can't be used in other project and when we try to use that e get violation error

Reply:
@anuragthehatter thanks for the explaination...LGTM.